src/progress-indicator/progress-indicator.component.ts

```diff
@@ -48,10 +48,12 @@ import { Step } from "./progress-indicator-step.interface";
 					(click)="stepSelected.emit({ step: step, index: i })">
 					{{step.text}}
 				</p>
-				<p class="bx--progress-label" *ngIf="!step.tooltip" (click)="stepSelected.emit({ step: step, index: i })">{{step.text}}</p>
-				<p *ngIf="step.optionalText" class="bx--progress-optional">{{step.optionalText}}</p>
+				<div class="bx--progress-text">
```

Comment:
```suggestion
				<div class="bx--progress-text" *ngIf="!step.tooltip || step.optionalText">
```

If both are false, we shouldn't render an empty node.

Reply:
@Akshat55 , i think we might need this condition instead.

```
*ngIf="(!step.tooltip && step.optionalText) || step.optionalText"
```

Reply:
hmm..i don't think there is a cleaner way to avoid rendering empty node.